docs/quick.rst

```diff
@@ -48,7 +51,39 @@ from a dataset in the file:
 
 For more, see :ref:`file` and :ref:`dataset`.
 
+Appendix: Creating a file
++++++++++++++++++++++++++
+
+At this point, you may wonder how :code:`mytestdata.hdf5` is created. 
+We can create a file by setting the :code:`mode` to :code:`w` when 
+the File object is initialized. Some other modes are :code:`a` 
+(for read/write/create access), and 
+:code:`r+` (for read/write access). 
+A full list of file access modes and their meanings is at :ref:`file`. ::  
+
+    >>> import h5py
+    >>> import numpy as np
+    >>> f = h5py.File("mytestfile.hdf5", "w")
 
+The :ref:`File object <file>` has a couple of methods which look interesting. One of them is ``create_dataset``, which 
+as the name suggests, creates a data set of given shape and dtype ::
+
+    >>> dset = f.create_dataset("mydataset", (100,), dtype='i')
+
+Now we can flush the file to the disk
+
+    >>> f.flush()
+    >>> del f
+
+The File object is a context manager; thus we can create the file and
+manage the :code:`flush` operation with the following lines ::
```

Comment:
The file is closed at the end of the "with" block, not when the interpreter
exits.

On Fri, Aug 28, 2015 at 12:55 PM, Yu Feng <notifications@github.com> wrote:

> In docs/quick.rst
> <https://github.com/h5py/h5py/pull/610#discussion_r38232293>:
>
> >
> > +The :ref:`File object <file>` has a couple of methods which look interesting. One of them is ``create_dataset``, which
> > +as the name suggests, creates a data set of given shape and dtype ::
> > +
> > +    >>> dset = f.create_dataset("mydataset", (100,), dtype='i')
> > +
> > +Now we can flush the file to the disk
> > +
> > +    >>> f.flush()
> > +    >>> del f
> > +
> > +The File object is a context manager; thus we can create the file and
> > +manage the :code:`flush` operation with the following lines ::
>
> By reading the code (files.py) it seems to close the file at *exit*. But
> what happens if I use another with block on f?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5py/h5py/pull/610/files#r38232293>.
>